netdisco/discoverables/xiaomi.py

```diff
@@ -0,0 +1,34 @@
+"""Discover Xiaomi Mi Home (aka Lumi) Gateways."""
+from . import MDNSDiscoverable
+from ..const import ATTR_MAC_ADDRESS, ATTR_PROPERTIES
+
+
+# pylint: disable=too-few-public-methods
+class Discoverable(MDNSDiscoverable):
+    """Add support for discovering Xiaomi Gateway"""
+
+    def __init__(self, nd):
+        """Initialize the discovery."""
+        super(Discoverable, self).__init__(nd, '_miio._udp.local.')
+
+    def info_from_entry(self, entry):
+        """Return most important info from mDNS entries."""
+        info = super().info_from_entry(entry)
+
+        # Workaround of misparsing of mDNS properties. It's unclear
+        # whether it's bug in zeroconf module or in the Gateway, but
+        # returned properties look like:
+        # {b'poch': b'0:mac=286c07aaaaaa\x00'} instead of expected:
+        # {b'epoch': b'0', b'mac': '286c07aaaaaa'}
```

Comment:
Did you check this out? At least in my case avahi-browse shows that the device is returning those just fine:
```
=  wlan0 IPv4 lumi-gateway-v3_miio5203XXXX                  _miio._udp           local
   hostname = [lumi-gateway-v3_miio5203XXXX.local]
   address = [192.168.XXX.XXX]
   port = [54321]
   txt = ["mac=286c07XXXXXX" "epoch=1"]
```

python-mirobo uses also the same zeroconf lib, and it seems to work just fine with my gateway:
```
ServiceInfo(type='_miio._udp.local.', name='lumi-gateway-v3_miio5203XXXX._miio._udp.local.', address=b'\xc0\xa8XX', port=54321, weight=0, priority=0, server='lumi-gateway-v3_miio52039961.local.', properties={b'mac': b'286c07XXXXXX', b'epoch': b'1'})
```

Reply:
Of course I did. I have a yeelight and a gateway, eveything is fine with the yeelight, and misparsed with the gateway.

Reply:
For reference, when I use Avahi Zeroconf Browser, I can see additional info when clicking on yeelink's entry, but when clicking on lumi's (i.e. gateway) entry I get:

> Error: org.freedesktop.Avahi.TimeoutError: Timeout reached

I may imagine that depends on various factors, e.g. firmware version used (and we should be prepared that newer firmware versions will break what we know worked before).